[27기 길도연][Mission 1,2,3,4,5,6,9] 필수 구현사항 및 추가 기능(7,8 제외) 구현 완료

sanha-kil opened the pull request:
## :: 최근 작업 주제 (하나 이상의 주제를 선택해주세요.)
- [x] 기능 추가
- [ ] 리뷰 반영
- [x] 리팩토링
- [x] 버그 수정
- [ ] 컨벤션 수정

<br />

## :: 구현 목표 (해당 브랜치에서 구현하고자 하는 하나의 목표를 설정합니다.)
- 인스타그램 로그인 페이지 및 메인페이지 클론 코딩

<br />

## :: 구현 사항 설명 (작업한 내용을 상세하게 기록합니다.)
1. 로그인 페이지 
  - [미션1] login.html 및 login.css 레이아웃 작성하였습니다.

2. 로그인 유효성 검사
  - [미션2] login.js를 통해 로그인 유효성 검사 구현하였습니다.
  - 삼항연산자를 사용하였습니다.
  - id, pw 각각의 input창을 getElementbyId로 불러들이고, 두 개의 이벤트리스너를 돌려 각각의 키업 이벤트가 발생할 때마다 검사를 진행하도록 했습니다.
  - id, pw input이 각각 한글자 이상 채워져있다면 로그인버튼의 disabled값에 false를 전달해 버튼을 활성화 시키고, 아닌 경우 true를 전달해 버튼을 비활성화 하도록 하였습니다.
  - [미션5] 아이디 형식 및 비밀번호의 길이를 통해 로그인 유효성 검사를 구현하였습니다.
  - 정규식을 이용하여 아이디 형식을 감별하고 비밀번호 5글자 이상이어야 로그인 버튼이 활성화 되도록 하였습니다.

3. 메인 페이지
  - [미션3] 메인페이지 main.html 및 main.css 레이아웃 작성하였습니다.

4. 피드 댓글 게시 기능
 - [미션4] 댓글 게시 기능 구현하였습니다.
 - 버튼 클릭 혹은 엔터키 keydown 이벤트 발생 시, 폼의 input란의 value를 댓글로 등록합니다.
 - [미션6] 피드 및 댓글의 좋아요 기능을 classList.toggle()을 이용해 구현 하였습니다.
 - 피드 댓글의 삭제기능을 구현하였습니다. confirm()을 이용하여 사용자에게 댓글 삭제여부를 물어보도록 구현하였습니다.

5. 반응형 스크린
 - [미션9] 가로 화면 크기에 따른 반응형 스크린을 구현하였습니다.
 - 미디어쿼리(@media screen)를 사용하여 가로길이 960px 이하일 경우 오른쪽 유저탭을 보이지 않도록 하였습니다.

6. form submit으로 인한 새로고침 수정
 - e.preventDefault를 이용하여 form의 submit시 새로고침되는 현상을 막았습니다.

7. 전체적인 리팩토링
 - CSS를 비롯한 코드들에 리팩토링을 적용하였습니다.
<br />

## :: 성장 포인트 (해당 기능을 구현하며 고민했던 사항이나 새로 알게된 부분, 어려웠던 점 등을 작성합니다.)
- 플렉스 컨테이너 내에서 메인페이지 허상의 레이아웃을 이용해 유저 추천 탭을 고정(display: fixed;)해보면서 창의적인 레이아웃 설계에 대해서 고민해 보았습니다.
- 삼항연산자 사용으로 더 깔끔한 코드의 표현이 가능해졌습니다.
- addEventListener를 통한 이벤트 관리를 능숙하게 할 수 있도록 노력하였습니다.
- 시멘틱 태그를 활용한 레이아웃 구성에 익숙해졌습니다.
- 리팩토링을 통해 코딩 컨벤션의 필요에 대해 이해하였습니다.
<br />

## :: 기타 질문 및 특이 사항
- 없습니다.

yojuyoon commented on students/13th/jeonghyunkim/js/login.js:
```diff
@@ -0,0 +1,19 @@
+const login = document.querySelector(".main_login");
+
+function LoginEvent() {
+  if (id.value.length > 0 && password.value.length > 0) {
```
&&연산자 활용 good! 👍

yojuyoon commented on students/13th/jeonghyunkim/style/main.css:
```diff
@@ -0,0 +1,291 @@
+body {
+  background-color: rgba(var(--b3f, 250, 250, 250), 1);
```
css 변수를 활용해서 rgba값을 할당해준 이유가 있을까요? css변수로 color 및 css속성을 적용할 때 공통된 color값을 가진 여러 요소에 적용시킬 때 사용하곤 합니다. 지금과 같은 상황에서는 rgba값으로 변경시켜줘도 무방할 듯 합니다. 만약 변수로 지정해주실 경우에는 공통된 css 속성을 선언한 뒤 그 변수명을 할당해주는 방법으로 사용해주세요!🙂
- 참고 링크입니다.
https://developer.mozilla.org/ko/docs/Web/CSS/Using_CSS_custom_properties

yojuyoon reviewed (commented):
안녕하세요 정현님! 전반적으로 기능구현이나 html, css 모두 잘 진행해주신 것 같습니다! 남은 기능 구현 및 리뷰 반영하여 리팩토링도 화이팅 해주세요! 😊👍

yojuyoon reviewed (commented):
기능 구현 잘 해주셨네요! 고생 많으셨습니다 정현님! 👍

sanha-kil committed:
11/10 branch first commit

sanha-kil committed:
ADD: 로그인 페이지/메인 페이지 레이아웃 및 CSS 작성
- 로그인페이지 작성 완료
- 메인페이지 헤더(네비게이션) 및 피드 작성

sanha-kil committed:
ADD: 메인 페이지 레이아웃 구성완료
- 전체적인 레이아웃 구성 완료
- main 섹션 유저정보 및 추천회원 탭 구현(position: fixed;를 통한 탭 고정)

sanha-kil committed:
ADD: 로그인 유효성 검사 구현

sanha-kil committed:
ADD: 메인페이지 댓글 추가 기능 구현
- 버튼 클릭 혹은 엔터키 누를 시 댓글 입력.
- 댓글 입력 후 input란의 value값 초기화하도록 구현하였습니다.

sanha-kil committed:
ADD: ID/PW 유효성검사, 댓글 좋아요/삭제 기능, 반응형 스크린 구현
- 정규식을 통한 ID 유효성검사 및 PW 길이 기반 유효성 검사구현으로 버튼 활성화/비활성화 구현하였습니다.
- classList.toggle()을 이용한 피드/댓글 좋아요 기능과 confirm()을 이용한 댓글 삭제기능을 구현하였습니다.
- 미디어쿼리를 이용하여 화면 가로길이 960px이하일 때 오른쪽 탭을 보이지 않게하는 반응형 스크린을 구현하였습니다.

sanha-kil committed:
Refactor: CSS 파일 전체적인 리팩토링

sanha-kil committed:
Fixed : submit으로 인한 초기화 fix
- form submit시 화면 새로고침으로 인한 초기화 방지
- preventDefault를 이용하여 새로고침을 막음
